Add stock / quantity replenish (refill) function for consumables

veenone opened the pull request:
# Description

This branch extends the functionality to add stock number of consumables by having replenishment process instead of updating it from edit button. additionally new table is created to store the historical trace of each replenishments

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] Test A :
- [ ] Test B : 

**Test Configuration**:
* PHP version: 7.4.27
* MySQL version : 10.4.22
* Webserver version: Apache 2.4.52
* OS version: Windows 10


# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# screens

## consumable table - new button for replenishment
![image](https://user-images.githubusercontent.com/3839381/159531019-cfadf2b6-4a7f-4de3-8c33-1c94b77321fb.png)

## replenish page
![image](https://user-images.githubusercontent.com/3839381/159531178-80fecf56-2430-47f7-9e4b-16ca7a7b0d44.png)

## consumable view
![image](https://user-images.githubusercontent.com/3839381/159531419-4a809c4c-7c03-4164-8fe1-66cc15a11ef3.png)
- this view is still on progress.. adding tabs to switch between consumable users & replenish history
- still finding out how to get the data from table and put it in the table 🤔

veenone committed:
- updating validator to use localized strings
- adding new localized string for validation on some languages
rebase branch with the latest commit on develop branch

veenone committed:
- initial implementation of the feature
- implements: consumable stock replenishment
- initial update for consumable view

veenone committed:
adding new files
adding database migrations/2022_03_22_114700_create_consumables_stock.php

veenone committed:
add localization string used in views

veenone committed:
fix consumable view

TODO: find the way to attach consumable replenish to user view

veenone committed:
missed to add new assignment file

veenone committed:
- replenish history view implemented
- updating localization string
- TODO: findout how to attach the activity to observer so can be recorded in the log

veenone committed:
implementing logaction in controller
NOTE: not the correct location I think, will be moved once I know how to do it correctly

veenone committed:
fix field typo in migration script

veenone commented:
updated from last commit 

## consumables view is working now
### checkout history
![image](https://user-images.githubusercontent.com/3839381/160159409-5a3675c7-b82a-47a0-a14a-45bc218bbadc.png)

### replenish history
![image](https://user-images.githubusercontent.com/3839381/160159561-4b03cba1-6f53-4772-9e44-8372dd0203fe.png)

## user view has 2 informations now
![image](https://user-images.githubusercontent.com/3839381/160159767-feddb3a4-9fad-4d7b-9cc7-d763f17ae114.png)

## recent activity for replenish added
![image](https://user-images.githubusercontent.com/3839381/160160033-bc8dad0d-16dd-49e4-901d-6686897ceaef.png)


Next:
- Add field for order number : done
- Add field for document / image upload (?)

veenone committed:
implement order number (placement for PO/Doc nbr)

veenone commented:
updated from last commit

### implements order number
## during replenish
![image](https://user-images.githubusercontent.com/3839381/160218636-1c432021-7474-4201-b33b-3d95dd8819f0.png)

## view
![image](https://user-images.githubusercontent.com/3839381/160218669-63887254-c9be-4a3c-a98d-3c5d4205433e.png)

##user
![image](https://user-images.githubusercontent.com/3839381/160218692-1e06e483-19a2-444a-b56a-2236a17bcee2.png)

veenone committed:
basic implementation for file download done

veenone committed:
file download is working now
cleanup forgotten consumables layout

veenone commented:
File download has been updated

![image](https://user-images.githubusercontent.com/3839381/160757238-55825931-8140-4788-af14-ed0c3143c612.png)

veenone committed:
code cleanup & updating replenish.blade.php layout

veenone committed:
forgot to change translation string for replenish note

veenone commented:
updating replenish page to include consumable description
![image](https://user-images.githubusercontent.com/3839381/162608871-f32e05e0-cd16-40cb-9aae-ed54e9076906.png)

veenone committed:
adding conditional checking if attachment is not provided during replenish

veenone committed:
adding missing field in migration db & refactoring variables to be snake case

veenone committed:
fix view layout on consumable & user

veenone commented:
updated user view on consumable : 

![image](https://user-images.githubusercontent.com/3839381/181934524-a54fc48f-00b0-4ef8-b087-54c5f5a0a7e6.png)

uberbrady commented on .gitignore:
```diff
@@ -65,3 +65,9 @@ _ide_helper_models.php
 /.phplint-cache
 storage/ldap_client_tls.cert
 storage/ldap_client_tls.key
+composer.lock
```
We definitely do want these `composer.lock` files checked-in; it's the best way to confirm that we have exactly the versions of various libraries that we want.

uberbrady commented on .gitignore:
```diff
@@ -65,3 +65,9 @@ _ide_helper_models.php
+en-GB
+en-ID
+es-ES
+es-MX
```
I'm not sure why these are added to `.gitignore` but we probably don't want these entries added there.

snipe replied:
Definitely don't want those there - the language files are part of the repo and shouldn't be ignored

uberbrady commented on app/Events/ConsumableCheckedOut.php:
```diff
@@ -0,0 +1,37 @@
+<?php
```
This looks pretty solid and I think it could be a good idea, but I wonder if `CheckoutableCheckedOut` could be extended, instead, to include an optional quantity (maybe with a default of `1`)? Or maybe there's not enough overlap between these two different sets of functionality...not sure.

I can imagine a lot of other 'checkoutable' things needing to include a quantity so I'd love it if we could extend there, instead of making this new event.

Still not sure on that, though - and I can be convinced.

veenone replied:
I will look at it

veenone replied:
this file has been updated

uberbrady commented on app/Http/Controllers/Api/ConsumablesController.php:
```diff
@@ -236,6 +252,56 @@ public function getDataView($consumableId)
+        $this->authorize('view', Consumable::class);
+        
+        $rows = [];
+        $consumable->consumableReplenishAssignments;
```
I don't think you need this line; I think the `foreach()` below will execute the query for you.

veenone replied:
ok I will look at it

veenone replied:
line updated

uberbrady commented on app/Http/Controllers/Consumables/ConsumableReplenishController.php:
```diff
@@ -0,0 +1,165 @@
+            $extension =  $request->file('file')->getClientOriginalExtension();
+            $file_name = 'consumable_replenish-'.$consumable->id.'-'.str_random(8).'-'.str_slug(basename($file->getClientOriginalName(), '.'.$extension)).'.'.$extension;
+        
+            if ($extension=='svg') {
```
Excellent! Thanks for sanitizing this SVG :)

snipe replied:
Don't we already have a form request that handles resizing and sanitizing images?

uberbrady commented on app/Http/Controllers/Consumables/ConsumableReplenishController.php:
```diff
@@ -0,0 +1,165 @@
+        isset($file_name) ? $file_name : $file_name = ''; 
+
+        $consumable->replenishusers()->attach($consumable->id, [
+            'consumable_id' => $consumable->id,            
```
Not 100% sure we need this part, as it's part of the `attach()`?

veenone replied:
removed now

uberbrady commented on app/Http/Controllers/Consumables/ConsumableReplenishController.php:
```diff
@@ -0,0 +1,165 @@
+        event(new ConsumableCheckedOut($consumable, Auth::user(), $consumable->qty, $request->input('note'), $request->input('checkout_qty')));
+
+        // Updating quantity to consumable
+        DB::table('consumables')
```
Since we already have a `$consumable` Model element, we should be able to say `$consumable->qty = $addedquantity` and then `$consumable->save()` - though you could shorten that to `$consumable->update(['qty' => $addedquantity])` if you wanted. We just prefer to avoid using `DB::table()` constructs if there's any way we can.

veenone replied:
I will update it and check it further

veenone replied:
DB::table() constructs has been removed and updated

uberbrady commented on app/Http/Controllers/Consumables/ConsumablesController.php:
```diff
@@ -191,6 +219,28 @@ public function show($consumableId = null)
+    }
+
+     /**
+     * Return a view to display component information.
```
typo here? should say 'consumable'?

veenone replied:
ah, bad copy & paste .. I will fix it

uberbrady commented on composer.lock:
```diff
@@ -1,13192 +0,0 @@
-{
```
We definitely still want this file

veenone replied:
will review my gitignore further..

uberbrady commented on database/migrations/2022_03_22_114700_create_consumables_stock.php:
```diff
@@ -0,0 +1,38 @@
+     */
+    public function up()
+    {
+        Schema::create('consumables_stock', function (Blueprint $table) {
```
I feel like the way this table is used is that it really is a measure of `Replenishments` so maybe we should rename it to that?

veenone replied:
noted

veenone replied:
table renamed to consumables_replenishments now

uberbrady commented on public/web.config:
```diff
@@ -23,5 +23,12 @@
         <remove fileExtension=".woff2" />
         <mimeMap fileExtension=".woff2" mimeType="application/x-font-woff2" />
     </staticContent>
+        <directoryBrowse enabled="true" />
```
This probably isn't part of this change. Happy to take it as a separate PR if it's something we need though.

veenone replied:
yup sorry this file has been included by mistake. will revert it  to the original one

uberbrady commented on resources/lang/es-VE/validation.php:
```diff
@@ -1,138 +0,0 @@
-<?php
```
We don't want to remove these language files - best to just make your additions to the `en` language files and leave the rest alone - they'll get automatically updated as our translators find the new translation strings to translate.

veenone replied:
Will do the update

uberbrady reviewed (changes requested):
This definitely seems interesting. It looks to me like we are adding the concept of a 'replenishment' which should adjust the number of available consumables, but also has data like who replenished it, and any notes about the replenishment, and associated attached files.

If we can clean up some of the extraneous or inadvertently deleted files in here, that would definitely help us out. I have a colleague (@inietov) who is going to work on testing this change; and he may have some feedback for you as well.

veenone committed:
updates based on review

- revert gitignore for some files
- update some lines of codes based on review
- update database table name to consumables_replenishments

veenone committed:
missed to save the changes on last commit

veenone commented:
Hi All,

changes have been made. looking forward to your review

adamboutcher commented:
This would make snipeit's consumables actually work as we'd expect... Just waiting on this merge.

Seamus-Mccartin commented:
@snipe This has been a requested feature for a couple years now, and seems to have been stuck waiting for review for a year now. Any update?

adamboutcher commented:
The same mechanic could be used in the Accessories too.

They really should be simple such as an item with a counter that can go up with orders and down with checkouts.

Solverz-0 commented:
> The same mechanic could be used in the Accessories too.
> 
> They really should be simple such as an item with a counter that can go up with orders and down with checkouts.

I disagree that this should be used for accessories, as if more accessories are purchased then the total quantity should increase and checked out accessories can be checked back in, whereas with consumables the checked out items are well "consumed" and therefore cannot be checked back in but should be replenished when more consumables are ordered without the total quantity increasing, like this new feature would add.

adamboutcher commented:
> > The same mechanic could be used in the Accessories too.
> > They really should be simple such as an item with a counter that can go up with orders and down with checkouts.
> 
> I disagree that this should be used for accessories, as if more accessories are purchased then the total quantity should increase and checked out accessories can be checked back in, whereas with consumables the checked out items are well "consumed" and therefore cannot be checked back in but should be replenished when more consumables are ordered without the total quantity increasing, like this new feature would add.

However currently adding new accessories, you're required to add your quantity to the current quantity which is really open to human error. It also asks for an order number but only allows a single accessory ID to a single order, where as it's really a Many (Orders/Checkin) to One (Accessory/Consumable) to Many (Checkouts) relationship. However the difference between consumables is that you simply never checkin once its checked out.

afranchi74 commented:
@snipe Has this feature been released? Any update?

etaxi341 commented:
@snipe when will this be merged? This would make snipe-it so much more usable. Currently consumables and accessories are really useless without this feature

zeehaque commented:
@snipe pls implement 😢

adamboutcher commented:
> @snipe when will this be merged? This would make snipe-it so much more usable. Currently consumables and accessories are really useless without this feature

I agree, especially the consumables. The current data model just doesn't work for something that's designed to disappear.

afranchi74 commented:
please lets go ahead with this features

afranchi74 commented:
@snipe when will this be merged? This would make snipe-it so much more usable. Currently consumables and accessories are really useless without this feature

snipe committed:
Prevent ludicrious amount of min_qty

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed incorrect jquery validation class

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Capture the log meta for what was changed

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added history tab

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use better icon for deleting

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use the observer to delete associated files and detach users

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Handle deletions on the observer

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15071 from snipe/fixes/small_consumables_optimizations

Small consumables optimizations

snipe committed:
Merge pull request #15069 from TelecomsSansFrontieres/fix_print-all-assets

Fixed inconsistent attributes order in Print All Assigned report

snipe committed:
Add @FlorentDotMe as a contributor

snipe committed:
Bumped version

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #14722 from Godmartinz/unaccepted_reminder_command

Adds a command to resend acceptance emails

marcusmoore committed:
Allow setting a prefix for Livewire's update and asset urls

snipe committed:
Merge pull request #15073 from marcusmoore/feature/livewire-url-prefix

Allow setting a prefix for Livewire's update and asset urls

akemidx committed:
missing translation

snipe committed:
Fixed return type

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added unaccepted asset test

Signed-off-by: snipe <snipe@snipe.net>

dbakan committed:
add test to ensure artisan optimize works

dbakan committed:
fix routes and tests

snipe committed:
Use app helped instead of facade

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use app()->getLocale() to determine imported user’s language

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Do not update users who already exist since that locale may have been overirrden manually

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed field name

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15082 from dbakan/fix/optimize-command

Fixed #10224: fix route names for optimize command

snipe committed:
Add @dbakan as a contributor

snipe committed:
Merge branch 'develop' of https://github.com/snipe/snipe-it into develop

snipe committed:
Small layout tweaks to oauth page

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Re-use existing string

Signed-off-by: snipe <snipe@snipe.net>

mauro-miatello committed:
Added EULAs in print user's assets

Added EULAs when we want to print all assets assigned to a users and ask him/her to sign.

carlkhenderson commented:
Just started using snipe IT and this has come up in the first day of use as an issue. This fix would address our concerns.

dependabot[bot] committed:
Bump codacy/codacy-analysis-cli-action from 4.4.1 to 4.4.5

Bumps [codacy/codacy-analysis-cli-action](https://github.com/codacy/codacy-analysis-cli-action) from 4.4.1 to 4.4.5.
- [Release notes](https://github.com/codacy/codacy-analysis-cli-action/releases)
- [Commits](https://github.com/codacy/codacy-analysis-cli-action/compare/v4.4.1...v4.4.5)

---
updated-dependencies:
- dependency-name: codacy/codacy-analysis-cli-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

snipe committed:
Corrected return type

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed #15094 - wrong translation string for model on checkout

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15099 from snipe/fixes/15094_wrong_translation_string

Fixed #15094 - wrong translation string for model on checkout

snipe committed:
Added select2 language files

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added language support for select2

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed translation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Handle localization better, clearer indenting on addresses

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Translated “seat” for license seat view

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Translate importer header

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Translate import button

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added basic jquery validation translations

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Translations for license checkout errors

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
More consistent UI for indented checkboxes

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Cleared up language

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Missing string

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated string

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use translated strings for action log meta

This is only partially complete and I mostly hate it

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15085 from snipe/small_tweaks_to_oauth_page

Small layout tweaks to oauth page

snipe committed:
Merge pull request #15089 from snipe/dependabot/github_actions/develop/codacy/codacy-analysis-cli-action-4.4.5

Bump codacy/codacy-analysis-cli-action from 4.4.1 to 4.4.5

snipe committed:
Merge pull request #15084 from snipe/fixes_ldap_sync_locale

Fixes #15067 - updated ldap sync locale to use `app()->getLocale()`

snipe committed:
Merge pull request #15075 from akemidx/checkintranslation

FIXED: Missing Translation on a Tooltip

marcusmoore committed:
Fix test namespaces

marcusmoore committed:
Clear the config and route caching done by optimize command after test

Glukose1 commented:
Hey @veenone Are you still interested in this? Because there are probably a lot of conflicts to be resolved by now so even if Snipe see this PR again and wants to merge it you would probably have to resolve the conflicts.

snipe committed:
Added new form fields

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added qty to email

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15111 from marcusmoore/fixes/optimize-test

Undo `optimize` command in test after it is completed

snipe committed:
Updated route

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added SubstituteBindings

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added accessory checkout request

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added store accesstory form request

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated validation strings

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added qty to email notification

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Cleaned up controllers, use form requests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added form requests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added numCheckedOut method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
More refactoring

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed int

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed debugging

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added translation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added qty

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed debugging

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Apply the optimize fix

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use from routes for posting

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Test passing now - I hope

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added back in missing validation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added more back :(

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge branch 'develop' into checkout_multiple_accessories

snipe committed:
Merge pull request #15088 from mauro-miatello/develop

Added EULAs in print user's assets

r-xyz committed:
Implement `PHP_UPLOAD_LIMIT` in Alpine images.

veenone commented:
Hi @Glukose1,
Yeah I'm still interested with this feature. In fact I'm currently still using it in my own setup for my own purpose.

I'll try to take a look on it and maybe sync the code to the latest develop branch. I will take my time to recall my implementation and see the gap with the latest version

snipe committed:
Updated count

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed response in form request

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15114 from snipe/checkout_multiple_accessories

Checkout multiple of an accessory in one checkout

snipe committed:
Merge pull request #15115 from r-xyz/alpine-upload-limit

Improved Docker environment settings: implement `PHP_UPLOAD_LIMIT` for Alpine images

snipe committed:
Add @r-xyz as a contributor

snipe committed:
Merge pull request #15110 from marcusmoore/fixes/test-namespace-fixes

Fixed a coupled test namespaces

snipe committed:
Merge branch 'develop' into localizations/more_strings

snipe committed:
Merge pull request #15101 from snipe/localizations/more_strings

More localizations

snipe committed:
Merge branch 'develop' of https://github.com/snipe/snipe-it into develop

snipe committed:
Updated strings

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15120 from snipe/updated_localizations

Updated strings

snipe committed:
Merge branch 'develop' into develop

snipe committed:
Merge pull request #14530 from DrekiDegga/develop

Added #14426: Makes all Manufacturer links dynamic, not just warranty lookup

snipe committed:
Add @DrekiDegga as a contributor

snipe committed:
Merge branch 'develop' of https://github.com/snipe/snipe-it into develop

snipe committed:
Updated url checks

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed #15121 - bulk delete restore logging

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15122 from snipe/fixes/15121_bulk_delete_and_restore

Fixed #15121 - bulk delete restore logging

snipe committed:
Added bulk asset delete test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed notification use statement

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15123 from snipe/tests/added_bulk_delete_assets

Added tests for bulk asset deletion and restore

snipe committed:
Fixed use statement

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15124 from snipe/fixes/path_for_crypt_in_importer

Fixed use statement for Crypt in the importer, removed unused statements

snipe committed:
Show existing images on user edit page

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15125 from snipe/fixes/display_avatar_on_user_edit

Show existing images on user edit page

snipe committed:
Bumped hash

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Remove unusued default avatar

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed unused default avatar

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added default avatar to seeder

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Skip deleting default avatar from disk on delete

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Ignore the test uploads directory

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Split out the image deletion into a separate method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added restore from demo method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added option to restore from default demo

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Cleaned up

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use new deleting method in ImageUploadRequest

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed incorrect signature

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
One more bad sig

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed restoreDefaultAvatar method

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated language

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated strings

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use other default avatar, not in the avatars directory

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15127 from snipe/fixes/small_default_avatar_tweaks

Fixes #15076 - Removes ability to remove the default avatar from disk

snipe committed:
Fixed default

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Nicer consumables layout

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Few more tweaks

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added clone to consumables

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added more generic buttons with variables

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
More updates

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15128 from snipe/fixes/nicer_consumables_layout

Nicer consumables layout

snipe committed:
Hides the “Disable Other Login Mechanisms” option via env

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed env part, just check for the remote user login

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Pulled from the app.php as well

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15132 from snipe/fixes/disable_remote_login_force_via_env

Hides the “Disable Other Login Mechanisms” option via env

snipe committed:
Load the english file again in case BS table doesn’t have a translation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15133 from snipe/fixes/load_english_separately_for_table_locale

Load the english file again in case BS table doesn’t have a translation

snipe committed:
[Snyk] Upgrade bootstrap-table from 1.22.5 to 1.23.0 #15131

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15134 from snipe/upgrade_bs_table

[Snyk] Upgrade bootstrap-table from 1.22.5 to 1.23.0 #15131

snipe committed:
Remove cookie serialization

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Customize cookie name

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Make the passport cookie name configurable

Signed-off-by: snipe <snipe@snipe.net>

uberbrady committed:
Improve .env file checking to not validate SSL certificates

snipe committed:
Added route param to message

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Switch to nb-NO from no-NO

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Migration

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated translations

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15144 from snipe/localization/updated_strings_2024_07_23

Updated translations

snipe committed:
Fixed test to use nb-NO

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15143 from snipe/fixes/no-NO-language

Switch to nb-NO from no-NO for Norwegian

spencerrlongg committed:
Merge branch 'develop' into bug/sc-24884

spencerrlongg committed:
$id to $asset->id (thanks tests)

spencerrlongg committed:
quick push

uberbrady committed:
Mark test as Incomplete, and downgrade error back to debug for tests

uberbrady committed:
Merge branch 'develop' into fix_setup_ssl_check

spencerrlongg committed:
test added, permission fixed

snipe committed:
Merge pull request #15147 from uberbrady/fix_setup_ssl_check

Fix setup ssl check

snipe committed:
Fixed string

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #14458 from spencerrlongg/bug/sc-24884

Add Form Request and Tests for Update Asset API Method

snipe committed:
Merge pull request #15136 from snipe/fixes/cookie_serialization

Remove cookie serialization

marcusmoore committed:
Handle empty strings for dates by converting them to null

snipe committed:
Fixed rule

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15148 from marcusmoore/fixes/dates-in-user-import

Fixed start_date and end_date in user importer

spencerrlongg committed:
test works, helper still needs work. pushing to check something

spencerrlongg committed:
ugh

spencerrlongg committed:
this works

spencerrlongg committed:
Merge branch 'develop' into bug/check_requiredness

spencerrlongg committed:
conflict

spencerrlongg committed:
ok NOW it should be fine

marcusmoore committed:
Register anonymous blade component namespace

snipe committed:
Merge pull request #15149 from spencerrlongg/bug/check_requiredness

Fixed: Requiredness Check Bug

snipe committed:
Early return for requiredness

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed env var for backup config - related to #14964

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15153 from snipe/fixes/mail_backup_config

Fixed env var for backup config - related to #14964

snipe committed:
Fixes #15103 - Added EOL and audit into to user profile assets

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15154 from snipe/fixes/15103_show_eol_to_users

Fixes #15103 - Added EOL and audit into to user profile assets

snipe committed:
Cleaned up UI on asset view

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15158 from snipe/fixes/ui_cleanup_for_asset_view

Cleaned up UI on asset view

snipe committed:
Removed the banner since we already warn

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Use route model binding on clone, removed unneeded route

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Allow cloning of deleted assets

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed double semicolon

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15160 from snipe/fixes/allow_cloning_of_deleted_assets

Allow cloning of deleted assets

snipe committed:
Fixed pluralization

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added check for purchase_date

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added check for purchase date

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed test no longer needed due to validation

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15161 from snipe/fixes/500_when_depreciation_is_active_but_no_purchase_date

Fixes 500 when depreciation is active but no purchase date

snipe committed:
Added assets endpoint for locations

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
One more test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15162 from snipe/fixes/added_location_assets_endpoint

Added assets endpoint for locations

snipe committed:
Baremetrics BMPay breaks with CSP turned [sc-25011]

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15150 from marcusmoore/chore/sc-26113

Registered custom anonymous blade component directory

snipe committed:
Use fully qualified use statements

Signed-off-by: snipe <snipe@snipe.net>

uberbrady committed:
Better handle older SQL dumps that got created in Windows format

snipe committed:
Merge pull request #15168 from uberbrady/improve_restore_sanitization

Improve restore sanitization - Fixes [sc-24840]

uberbrady committed:
Create intermediate directories on restore if needed

snipe committed:
Fixed BYOD label

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15169 from uberbrady/create_directories_when_needed_on_restore

Create intermediate directories on restore if needed - Fixes [SC-25950]

uberbrady committed:
Dynamically iterate through the skin listing to build skins

snipe committed:
Use blade component for redirect option

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Refactor helper method for redirection

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Add status messages to redirect

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Removed logging

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Updated test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added created_by to license view

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed back button

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Pulled use session

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added more redirects

Signed-off-by: snipe <snipe@snipe.net>

uberbrady committed:
Use a similar loop for the minification step for CSS

snipe committed:
Added parameters

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added blade component to checkin/checkout pages

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added redirect to checkin/checkout controllers

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Made dropdown wider

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Fixed logging for component

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Very basic checkin tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added more tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
More tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added more tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Renamed test

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
And more tests

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Merge pull request #15172 from snipe/features/blade_component_for_submit

Use blade component for submit redirect on asset edit/create

snipe committed:
Merge pull request #15170 from uberbrady/simplify_css_skin_building

Dynamically iterate through the skin listing to build skins

snipe committed:
Use translated string for restore

Signed-off-by: snipe <snipe@snipe.net>

snipe committed:
Added qr code back

Signed-off-by: snipe <snipe@snipe.net>

veenone committed:
rebase branch with develop branch

veenone committed:
small adjustments after rebase

veenone committed:
deactivating badge on replenishconsumables history tab

veenone commented:
rebase to the last develop branch

afranchi74 commented:
@veenone please double check because "some checks were not successful".
Thanks in advance

afranchi74 commented:
@veenone please double check because "some checks were not successful".
Thanks in advance

lubna201 commented:
Any update on this?
